docs/progress.md

```diff
@@ -56,13 +60,13 @@ spec:
           - name: progress
             template: progress
     - name: progress
-      metadata:
-        annotations:
-          workflows.argoproj.io/progress: 0/100
       container:
         image: alpine:3.14
         command: [ "/bin/sh", "-c" ]
         args:
           - |
             for i in `seq 1 10`; do sleep 10; echo "$(($i*10))"'/100' > $ARGO_PROGRESS_FILE; done
 ```
+
+Please make sure you have granted `patch` privilege on `workflowtaskresults` resources for the executor
+to be able to update progress on the workflow. See https://argoproj.github.io/argo-workflows/workflow-rbac/
```

Comment:
```suggestion
You must set your [Workflow RBAC](workflow-rbac.md) properly for the executor to be able to update progress.
```

- relative links are used when referring to other docs
- links are preferred to be in-line and contextual, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#links)
- use more direct language, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)